### ISSUE TITLE
Wrong `file_position` reported under Python 3 when file has CRLF line endings

### ISSUE THREAD

**Issue by pedrolamas** [Klipper3d/klipper#5755]
We've had reports in Fluidd of incorrect progress being reported and only now we found why this was happening!

Klipper uses the python inbuilt `open` function here:

https://github.com/Klipper3d/klipper/blob/9f8f3cd77d97339bd825053b064ae4d08d5d6902/klippy/extras/virtual_sdcard.py#L180

This works just fine under python 2, however this function seems to just ignore the CR of files with CRLF line endings!

To prove this, I ran the following steps:

- Started a clean debian docker container with `docker run --rm -it debian bash`
- `apt update && apt install -y python2 python3`
- Created a very simple file with CRLF by running `echo -ne "pedro\r\nlamas">test.txt`

After the above, I ran `python2` and these commands:
```python
f = open('test.txt', 'r')
data = f.read(100)
f.close()
for x in range(len(data)):
    print(ord(data[x]))
```

Output from python2:
```text
112
101
100
114
111
13
10
108
97
109
97
115
```

I can see clearly the CR (code 13) on the above output.

Also, doing `len(data)` returns 12, which is what I expected.

Running the same on `python3`, and this is the output:

```text
112
101
100
114
111
10
108
97
109
97
115
```

We can see clearly that CR (code 13) on the above output is missing!

Also, doing `len(data)` will return 11 instead of 12.

Bottom line is that `open` function under python3 is ignoring CR chars, thus leading to an incorrect `file_position` being reported.

**Comment by pedrolamas** [comment]
From my tests, replacing that `open` with `io.open` seems to fix the issue in python3!

```diff
-    f = open(fname, 'r')
+    f = io.open(fname, 'r', newline='')
```
